libmscore/barline.cpp

```diff
@@ -566,7 +566,10 @@ printf("may drop\n");
                && segment()
                && segment()->isEndBarLineType());
             }
+#if (!defined (_MSCVER) && !defined (_MSC_VER))
+      // Prevent unreachable code warning in MSVC
```

Comment:
If that code is unreachable irrespectively of compiler used, remove it altogether